client/me/form-base/with-form-base.jsx

```diff
@@ -8,96 +8,54 @@ import { connect } from 'react-redux';
 /**
  * Internal dependencies
  */
-import user from 'calypso/lib/user';
-import userSettings from 'calypso/lib/user-settings';
 import { errorNotice, successNotice } from 'calypso/state/notices/actions';
+import QueryUserSettings from 'calypso/components/data/query-user-settings';
+import getUserSettings from 'calypso/state/selectors/get-user-settings';
+import getUnsavedUserSettings from 'calypso/state/selectors/get-unsaved-user-settings';
+import hasUnsavedUserSettings from 'calypso/state/selectors/has-unsaved-user-settings';
+import {
+	setUserSetting,
+	clearUnsavedUserSettings,
+	saveUserSettings,
+} from 'calypso/state/user-settings/actions';
+import { isUpdatingUserSettings } from 'calypso/state/user-settings/selectors';
 
 const withFormBase = ( WrappedComponent ) => {
 	class EnhancedComponent extends React.Component {
 		static displayName = `withFormBase(${ WrappedComponent.displayName || WrappedComponent.name })`;
 
 		state = {
 			redirect: false,
```

Comment:
We can also ditch that `redirect` and the entire component state, it doesn't seem that it's used anywhere any more.

Reply:
fixed via https://github.com/Automattic/wp-calypso/pull/48730/commits/a9a666eb81eb5dfe956e3e23d1d427cad787d5e5